### DISCUSSION TITLE
decimal precision of continuous variable

### DISCUSSION THREAD

**Discussion by joelguerrero** [snl-dakota/dakota#13] [Q&A]
Hi,

I am setting an optimization case, and I am wondering if it is possible to generate the continuous variable values with a given precision.  For instance, I am interested in generating the values with no more than 3 decimal points.

So far i am using the option --output-format '%.3f' with dprepro, and it is substituting the variables with the desired precision.  I am also saving the output with the desired precision. However, by looking at the file params.in I see that dakota still is using the complete precision. I also see that dakota is using the complete precision to evaluate the gradients.  

I wonder if it is possible to change the precision in the file params.in, so the gradients (and hessians) are evaluated using the desired precision? 

Thank you very much for your help,

joel

**Comment by jadamstephens** [comment]
Joel,

Sorry, no.

Some of Dakota's derivative-free local optimizers have a [variable_tolerance](https://dakota.sandia.gov/sites/default/files/docs/6.17.0-release/user-html/usingdakota/reference/method-mesh_adaptive_search-variable_tolerance.html) convergence criterion that may accomplish what you want.

Another goofy idea.. if you can live with a derivative-free optimizer, you could convert your variables to [discrete_design_range](https://dakota.sandia.gov/sites/default/files/docs/6.17.0-release/user-html/usingdakota/reference/variables-discrete_design_range.html#) variables that have the desired number of steps and scale them to floating point values in your driver. That is, if you wanted a continuous variable between 1 and 2 with precision of 0.001, you'd set up a discrete range variable 1000-2000 and divide by 1000.0 in your driver.

Adam
